OPHD/UI/IconGrid.cpp

```diff
@@ -86,8 +86,11 @@ void IconGrid::onMouseDown(EventHandler::MouseButton button, int x, int y)
 		return;
 	}
 
+	auto previousIndex = mSelectedIndex;
 	mSelectedIndex = translateCoordsToIndex(mousePoint - startPoint);
 
+	if (previousIndex == mSelectedIndex) { return; }
+
 	if (mSelectedIndex >= mIconItemList.size())
 	{
 		mSelectedIndex = constants::NO_SELECTION;
```

Comment:
Hmm, curious, what happens if the number of items in the `IconGrid` are updated? Particularly if items are removed.

I suppose `mSelectedIndex` would potentially be reset to `NO_SELECTION` during an update that changes the changes the list `size`. Though if that doesn't happen, this function may now exit before the `size` clamp, leaving `mSelectedIndex` set to an invalid out-of-bounds index.

Might be better to change the early return check to instead be a check around the `raiseChangedEvent()` call. That might also be a little more intuitive, as it would more directly address the problem at hand.

Reply:
Well, the index will always refer to an integer, if we want to cover all bases and make sure the same item is selected, we'd need to translate it into whatever that integer is pointing to and compare it to whatever the incoming value is pointing to.
But I don't think that knowledge exists at the IconGrid. Or is that what the selection_meta(int selectionMetaValue is for?